src/main/java/com/faforever/client/replay/OnlineReplayVaultController.java

```diff
@@ -182,16 +194,35 @@ private void refresh() {
 
   public void onMoreNewestButtonClicked() {
     enterSearchingState();
-    replayService.getNewestReplays(TOP_MORE_ELEMENT_COUNT).thenAccept(this::displaySearchResult);
+    displayReplaysFromSupplier(() ->replayService.getNewestReplays(TOP_MORE_ELEMENT_COUNT,currentPage++));
   }
 
   public void onMoreHighestRatedButtonClicked() {
     enterSearchingState();
-    replayService.getHighestRatedReplays(TOP_MORE_ELEMENT_COUNT).thenAccept(this::displaySearchResult);
+    displayReplaysFromSupplier(() ->replayService.getHighestRatedReplays(TOP_MORE_ELEMENT_COUNT,currentPage++));
   }
 
   public void onMoreMostWatchedButtonClicked() {
     enterSearchingState();
-    replayService.getMostWatchedReplays(TOP_MORE_ELEMENT_COUNT).thenAccept(this::displaySearchResult);
+    displayReplaysFromSupplier(() ->replayService.getMostWatchedReplays(TOP_MORE_ELEMENT_COUNT,currentPage++));
+  }
+
+  public void onLoadMoreButtonClicked(ActionEvent actionEvent) {
+    currentSupplier.get()
+        .thenAccept(replays -> displaySearchResult(replays,true));
+  }
+
+  private void displayReplaysFromSupplier(Supplier<CompletableFuture<List<Replay>>> mapsSupplier) {
+    currentPage=1;
+    this.currentSupplier = mapsSupplier;
+    mapsSupplier.get()
+        .thenAccept(this::displaySearchResult)
+        .exceptionally(e -> {
+          notificationService.addNotification(new ImmediateNotification(i18n.get("errorTitle"),
+              i18n.get("vault.replays.searchError"), Severity.ERROR, e,
+              Collections.singletonList(new DismissAction(i18n))));
+          enterShowroomState();
+          return null;
+        });
   }
```

Comment:
This file needs to be formatted correctly, too. Probably every file in the PR :D

Reply:
ok